packages/components/src/lib/MarkdownViewer.tsx

```diff
@@ -25,20 +30,47 @@ function rehypeInlineCodeProperty() {
   };
 }
 
+type CustomComponents = Components & {
+  squiggleplayground: React.ComponentType<{ children: string }>;
+  squiggleeditor: React.ComponentType<{ children: string }>;
+};
+
 type MarkdownViewerProps = {
   md: string;
   textSize: "sm" | "xs";
   textColor?: "prose-stone" | "prose-slate";
   className?: string;
   backgroundColor?: string;
 };
+
 export const MarkdownViewer: React.FC<MarkdownViewerProps> = ({
   md,
   className,
   textColor,
   textSize,
   backgroundColor = "bg-slate-50",
 }) => {
+  const CustomSquigglePlayground = ({ children }: { children: string }) => {
+    const trimmedChildren = trimNewlines(children);
+    const numLines = (trimmedChildren.match(/\n/g) || []).length + 1;
+    const linesToShow = Math.min(Math.max(numLines, 4), 30);
+    const height = 31 + linesToShow * (textSize === "sm" ? 20 : 18); // This is fairly hacky, but it works for now.
```

Comment:
In the future, we could expose something like [contentHeight](https://codemirror.net/docs/ref/#view.EditorView.contentHeight) (or maybe some other prop, I'm not sure, but there must be one that corresponds to what we need) from `<CodeEditor>` through the imperative handle, and then forward that in `<SquigglePlayground>`, but I understand that'd be more work, and we'd still have to account for the menu height.

Note that this won't work correctly with line wrapping, btw.

Reply:
Good point. I definitely agree that we could eventually use better solutions, this is pretty hacky.